### PR TITLE
Add github action to check urls are valid

### PR DESCRIPTION
**Pull request by xadupre**
### Description
This action checks that urls in the code or in the documentation are valid.

### Motivation and Context
It avoids having broken urls.

### CODE REVIEW

File: .github/workflows/check_urls.yml
Comment (jcwchen) [inline .github/workflows/check_urls.yml:3]: Shall we at least test it once before merging to the main branch? For instance, add pull_request trigger first and then remove it before merge?

File: .github/workflows/check_urls.yml
Comment (justinchuby) [inline .github/workflows/check_urls.yml:22]: Trying to understand this: so it will fail after urls are fixed?

File: .github/workflows/check_urls.yml
Comment (justinchuby) [inline .github/workflows/check_urls.yml:13]:
```suggestion
      uses: urlstechie/urlchecker-action@xxxx
```

For security I would pin a hash

**Comment by justinchuby** [comment]
I didn't see it running in this pr. Do we need to configure triggers?

**Review by jcwchen — COMMENTED** [review]
Thanks for the update -- now this added CI was triggered in this PR, but according to the log, the CI did not fail when it found some invalid URLs.

File: .github/workflows/check_urls.yml
Comment (xadupre) [reply .github/workflows/check_urls.yml:26]: It does not fail if `force_pass: true`, with false, it fails if the urls is not found. I was hoping the find a trigger which checks the urls regularly but not at every PR.

Comment (justinchuby) [reply .github/workflows/check_urls.yml:26]: I see! I misread. Thanks.

**Review by justinchuby — APPROVED** [review]

File: .github/workflows/check_urls.yml
Comment (justinchuby) [reply .github/workflows/check_urls.yml:26]: @jcwchen fyi

File: .github/workflows/check_urls.yml
Comment (jcwchen) [inline .github/workflows/check_urls.yml:17]:
```suggestion
      uses: urlstechie/urlchecker-action@b643b43e2ac605e1475331c7b67247d242b7dce4 # v0.0.34
```

File: .github/workflows/check_urls.yml
Comment (jcwchen) [inline .github/workflows/check_urls.yml:29]:
```suggestion
      uses: urlstechie/urlchecker-action@b643b43e2ac605e1475331c7b67247d242b7dce4 # v0.0.34
```

File: .github/workflows/check_urls.yml
Comment (jcwchen) [inline .github/workflows/check_urls.yml:41]:
```suggestion
      uses: urlstechie/urlchecker-action@b643b43e2ac605e1475331c7b67247d242b7dce4 # v0.0.34
```

File: .github/workflows/check_urls.yml
Comment (jcwchen) [inline .github/workflows/check_urls.yml:14]:
```suggestion
    - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
```

File: .github/workflows/check_urls.yml
Comment (xadupre) [reply .github/workflows/check_urls.yml]: done

**Review by jcwchen — APPROVED** [review]

**Comment by justinchuby** [comment]
@gramalingam for approval

**Comment by justinchuby** [comment]
@onnx/sig-operators-approvers for approval

**Review by wschin — APPROVED** [review]

File: .github/workflows/check_urls.yml
Comment (justinchuby) [inline .github/workflows/check_urls.yml:7]:
```suggestion
```

File: .github/workflows/check_urls.yml
Comment (justinchuby) [inline .github/workflows/check_urls.yml:5]:
```suggestion
  push:
    branches: [ 'rel-*' ]
  schedule:
    # Run every month
    - cron:  '0 0 1 * *'
  workflow_dispatch:
```

**Review by justinchuby — APPROVED** [review]

**Pull request merged**
